libr/core/anal_vt.c

```diff
@@ -183,3 +183,40 @@ static void r_core_anal_list_vtables_all(void *core) {
 	r_list_free (vtables);
 }
 
+static rtti_struct* get_rtti_data (RCore *core, ut64 atAddress) {
+	ut64 bits = r_config_get_i (core->config, "asm.bits");
+	int wordSize = bits / 8;
+	ut64 BaseLocatorAddr = r_io_read_i (core->io, atAddress - wordSize, wordSize);
+	r_cons_printf ("Trying to parse rtti at 0x%08"PFMT64x"\n", BaseLocatorAddr);
+	return NULL;
+}
+
+RList* r_core_anal_parse_rtti (void *core, bool printJson) {
+	RList* vtables = search_virtual_tables ((RCore *)core);
+	RListIter* vtableIter;
+	RList* rtti_structures = r_list_new();
+	vtable_info* table;
+
+	if (vtables) {
+		r_list_foreach (vtables, vtableIter, table) {
+			rtti_struct* current_rtti = get_rtti_data ((RCore *)core, table->saddr);
+			if (current_rtti) {
+				current_rtti->vtable_start_addr = table->saddr;
+				r_list_append (rtti_structures, current_rtti);
+			}
+		}
+	}
+	r_list_free (vtables);
+	return rtti_structures;
+}
+
+R_API void r_core_anal_print_rtti (void *core) {
+	RList* rtti_structures = r_core_anal_parse_rtti (core, false);
+	RListIter* RTTIIter;
+	rtti_struct* curRTTI;
```

Comment:
you can use shorter names here, RTTI is implicit, just 'cur'. also use r2-indent, that will put the * away from the typedef